File: server/src/main/java/com/server/loan/calculator/model/InitialData.java
package com.server.loan.calculator.model;

import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;

@Data
@Builder
public class InitialData {
    private BigDecimal euriborRate;
    private String euriborDate;
    private BigDecimal bankInterestRate;
    private BigDecimal annualInterestRate;
    private BigDecimal contractFee;
    private BigDecimal registrationFee;
    private BigDecimal monthlyBankFee;
    private BigDecimal maxPropertyPrice;
    private BigDecimal minPropertyPrice;
    private BigDecimal defaultPropertyPrice;
    private BigDecimal minDepositPercent;
    private BigDecimal defaultInitialDeposit;
    private int defaultMortgagePeriod;
    private BigDecimal defaultSalary;
    private BigDecimal defaultFinancialObligation;
}

File: client/src/app/Types/InitialData.ts
export interface InitialData {
    // lastUpdatedOn: string,
    // euribor: number,
    // bankMargin: number,
    // minInitialDeposit: number,
    // minPriceOfProperty: number,
    // maxPriceOfProperty: number,
    // defaultPriceOfProperty: number,
    // contractFee: number,
    // bankFee: number,
    // mortgageRegistrationFee: number


    euriborRate: number,
    euriborDate: string,
    bankInterestRate: number,
    annualInterestRate: number,
    contractFee: number,
    registrationFee: number,
    monthlyBankFee: number,
    maxPropertyPrice: number,
    minPropertyPrice: number,
    defaultPropertyPrice: number,
    minDepositPercent: number,
    defaultInitialDeposit: number,
    defaultMortgagePeriod: number,
    defaultSalary: number,
    defaultFinancialObligation: number
}

// {
//     euriborRate: number,
//     euriborDate: string,
//     bankInterestRate: number,
//     annualInterestRate: number,
//     contractFee: number,
//     registrationFee: number,
//     monthlyBankFee: number,
//     maxPropertyPrice: number,
//     minPropertyPrice: number,
//     defaultPropertyPrice: number,
//     minDepositPercent: number,
//     defaultInitialDeposit: number,
//     defaultMortgagePeriod: number,
//     defaultSalary: number,
//     defaultFinancialObligation: number
// }